File: content/code/cpp/Pet_sol.cpp
#include <catch2/catch.hpp>

#include "pet.hpp"

TEST_CASE("Check my pets", "[pets]") {
  auto fido = Pet("fido");
  REQUIRE(fido.hunger() == 0);

  fido.go_for_a_walk();
  REQUIRE(fido.hunger() == 1);
}

File: content/code/julia/Pet_sol.jl
# create the mutable struct and test it
@testset "Test mutable struct" begin
    p = Pet("fido", 0)
    @test p.hunger == 0
    go_for_a_walk!(p)
    @test p.hunger == 1
    p.hunger = -1
    go_for_a_walk!(p)
    @test p.hunger == 0
end
